doc/sphinx/proofs/automatic-tactics/auto.rst

```diff
@@ -248,7 +230,7 @@ One can optionally declare a hint database using the command
 :cmd:`Create HintDb`. If a hint is added to an unknown database, it will be
 automatically created.
 
-.. cmd:: Create HintDb @ident {? discriminated}
+.. cmd:: Create HintDb @ident {? discriminated }
```

Comment:
By "implicit" you mean those defined by Coq aka "built-in"?  What is the default transparency in either case?

The description for `Hint Constants ...` is poorly worded, please check what I changed there.

Reply:
No, I mean what is said just above the `Create HintDb` description: "If a hint is added to an unknown database, it will be
automatically created." When databases are automatically created that way, without the explicit `Create HintDb`, they do not get the same default transparency (this is stupid, but now we need to warn users against it).

See this example:

```coq
Hint Resolve I : foo.
Print HintDb foo.
(*
Unfoldable variable definitions: none
Unfoldable constant definitions: none
*)
Hint Constants Transparent : foo.
Hint Variables Transparent : foo.
Print HintDb foo.
(*
Unfoldable variable definitions: all
Unfoldable constant definitions: all
*)
Hint Opaque Nat.add : foo.
Print HintDb foo.
(*
Unfoldable variable definitions: all
Unfoldable constant definitions: all except: Nat.add
*)
Create HintDb bar.
Print HintDb bar.
(*
Unfoldable variable definitions: all
Unfoldable constant definitions: all
*)
Hint Constants Opaque : bar.
Hint Variables Opaque : bar.
Print HintDb bar.
(*
Unfoldable variable definitions: none
Unfoldable constant definitions: none
*)
Hint Transparent Nat.add : bar.
Print HintDb bar.
(*
Unfoldable variable definitions: none
Unfoldable constant definitions: Nat.add
*)
```

Reply:
> "If a hint is added to an unknown database, it will be automatically created."

Not very well worded...it didn't immediately register that "it" refers to the unknown database.  Nor is the "optionally" in the previous sentence a good choice.  We recommend using `Create HintDb`, right?

Reply:
Shouldn't we move toward requiring use of `Create HintDb` to help save users from typos?

Reply:
How is this?

![image](https://user-images.githubusercontent.com/1253341/99175439-c293b080-26bf-11eb-98a5-1c463ba0c36c.png)

Reply:
> Shouldn't we move toward requiring use of `Create HintDb` to help save users from typos?

Absolutely. But there is another change which is that hint databases should get qualified names and I'm not sure if the two changes interact or not.

Your proposed text looks good but I would word the recommendation even more strongly: We recommend to always create hint databases with the Create HintDb command and to follow with the Hint Constants and Hint Variables commands so that your choice of default transparency is made explicit.